### PR TITLE
Fix build and Deb packaging issues

### PR DESCRIPTION
**Pull request by andrewbonney**
General lingering packaging issues, however the sed line is worth some extra review. It resolves missing debhelper requirements for the debian/rules file. I could have put a custom rules file into the debian directory, but as py2dsc puts so much stuff into this it didn't feel right. This may or may not be a better approach.

### CODE REVIEW

File: debian/postinst [hunk 1]
```diff
@@ -23,7 +23,6 @@ fi
 
 case "$1" in
     configure)
-        service apache2 reload || true
```
Comment (samdbmg) [inline debian/postinst:26]: Do you still need the surrounding `case` block? It doesn't do anything anymore!

File: rpm/python-mdnsbridge.spec [hunk 2]
```diff
@@ -5,6 +5,8 @@ License: 		Internal Licence
 Summary: 		mDNS to HTTP bridge service
 
 Source0: 		%{name}-%{version}.tar.gz
+Source1:		ips-api-mdnsbridge.conf
+Source2:		python-mdnsbridge.service
```
Comment (samdbmg) [inline rpm/python-mdnsbridge.spec]: How do these get copied into `build/rpm/SOURCES`? I can see a Makefile target for the source tarball, but not these config files.

**Review by samdbmg — DISMISSED** [review]
Couple of questions inline. Also:

- We have a Makefile target to create an RPM specfile (`rpm/mdnsbridge.spec`) and also specfile in the repository `rpm/python-mdnsbridge.spec` - we should probably have one or the other.
- The RPM build doesn't appear to work. Even if I take out that Makefile target, then rename the existing specfile to drop the `python-` prefix. We should probably document somewhere if it's known to be broken.
- I'm pretty sure Jenkins doesn't run the sed command... It calls `make dsc` which has `deb_dist` as a prereq, but that `sed` is in `make deb`
- The Python 3 deb fails to install alongside Python 2 because they conflict `/usr/bin/nmos-mdnsbridge` - only one package (the one matching the shebang line) should own the binary
- Nitpick: The `Jenkinkinsfile` comments at the top say it runs the Python 3 unit tests in tox, but it doesn't.
- `make test` fails because Python 3 unit tests are broken - perhaps they should be removed from the tox envlist?
- Should we even build a python3 package since it doesn't work?
```
 samn@ap-z440-7 >  python3
Python 3.5.2 (default, Nov 23 2017, 16:37:01)
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from mdnsbridge.mdnsbridge import *
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/project-mcuk/ap/ipp/samn/ipstudio/nmos-mdns-bridge/mdnsbridge/mdnsbridge.py", line 137
    print "*** mDNS ***"
                       ^
SyntaxError: Missing parentheses in call to 'print'
>>>
```

File: debian/postinst (hunk 1 above)
Comment (andrewbonney) [reply debian/postinst:26]: I was leaving that in as people tend to forget to re-add it if they make changes. Without it things can execute multiple times on one install.

File: rpm/python-mdnsbridge.spec (hunk 2 above)
Comment (andrewbonney) [reply rpm/python-mdnsbridge.spec]: Is this not covered by the lines under %files? I'm not totally familiar with these so it may not be.

Comment (samdbmg) [reply rpm/python-mdnsbridge.spec]: That refers to their location in the target filesystem - the `%files` list is a list of files owned by this package

**Comment by andrewbonney** [comment]
Hoping most of these are resolved now other than the RPM stuff. For the Python 3 issues I've stopped the packages building, but I've left the rest in place for now for consistency in how this has been handled elsewhere, and given that we do need to put the effort in to make this support Python 3.

I'll take a look at the RPM issues next, although I suspect that in this case and others the packages haven't worked for a while.

File: Makefile [hunk 3]
```diff
@@ -75,10 +75,10 @@ deb: source deb_dist $(DEBIANOVERRIDES)
...
 # START OF RPM SPEC RULES
 # If you have your own rpm spec file to use you'll need to disable these rules
-$(topdir)/rpm/$(MODNAME).spec: rpm_spec
-
-rpm_spec: $(topdir)/setup.py
-	$(PYTHON3) $(topdir)/setup.py bdist_rpm --spec-only --dist-dir=$(topdir)/rpm
+#$(topdir)/rpm/$(MODNAME).spec: rpm_spec
+#
+#rpm_spec: $(topdir)/setup.py
+#	$(PYTHON3) $(topdir)/setup.py bdist_rpm --spec-only --dist-dir=$(topdir)/rpm
```
Comment (samdbmg) [inline Makefile:81]: Nitpick: Delete outright?

File: Jenkinsfile [hunk 4]
```diff
@@ -25,6 +25,9 @@ pipeline {
         booleanParam(name: "FORCE_PYUPLOAD", defaultValue: false, description: "Force Python artifact upload")
         booleanParam(name: "FORCE_DEBUPLOAD", defaultValue: false, description: "Force Debian package upload")
     }
+    triggers {
+        upstream (upstreamProjects: "apmm-repos/nmos-common/master")
```
Comment (samdbmg) [inline Jenkinsfile:29]: 👍

**Review by samdbmg — COMMENTED** [review]
Looks good. I've had a play and got the RPM build sort-of working in Docker. Firstly the Makefile needs a tweak to copy the two missing config files (ips-api-mdnsbrige.conf and python-mdnsbridge.service) into `build/rpm/SOURCES` in much the same way it already does for the source tarball.
Also in the specfile, remove line 62 (tests don't get installed it seems) and the following line should be `%{python2_sitelib}/mdnsbridge-%{version}*.egg-info` - then it seems to build!

File: Makefile (hunk 3 above)
Comment (andrewbonney) [reply Makefile:81]: I might leave this in if you don't mind just to make it extra obvious we're intentionally not doing it that way.

Comment (samdbmg) [reply Makefile:81]: Sounds good

**Review by samdbmg — APPROVED** [review]
Looks good (noting that RPM is untested, but seems legit). Will leave you to do version bumps at your leisure.

**Pull request merged**